### PR TITLE
Edit pass on Azure SDK authN docs

### PR DESCRIPTION
**Pull request by scottaddie**
**Summary of changes**
- Update Azure AD text, links, & screenshots for rebranding
- Make `UsernamePasswordCredential` cautionary text more prominent
- Update the `DefaultAzureCredential` docs
- Remove VS Code guidance, as it's broken. Add azd guidance.
- Incorporate Acrolinx feedback
- Fix content validation errors
- Add missing metadata to include files

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/azure/sdk/authentication/additional-methods.md](https://github.com/dotnet/docs/blob/d37b0d56517527ae281455e61fba4fcad98d19e2/docs/azure/sdk/authentication/additional-methods.md) | [docs/azure/sdk/authentication/additional-methods](https://review.learn.microsoft.com/en-us/dotnet/azure/sdk/authentication/additional-methods?branch=pr-en-us-41972) |
| [docs/azure/sdk/authentication/azure-hosted-apps.md](https://github.com/dotnet/docs/blob/d37b0d56517527ae281455e61fba4fcad98d19e2/docs/azure/sdk/authentication/azure-hosted-apps.md) | [docs/azure/sdk/authentication/azure-hosted-apps](https://review.learn.microsoft.com/en-us/dotnet/azure/sdk/authentication/azure-hosted-apps?branch=pr-en-us-41972) |
| [docs/azure/sdk/authentication/index.md](https://github.com/dotnet/docs/blob/d37b0d56517527ae281455e61fba4fcad98d19e2/docs/azure/sdk/authentication/index.md) | [docs/azure/sdk/authentication/index](https://review.learn.microsoft.com/en-us/dotnet/azure/sdk/authentication/index?branch=pr-en-us-41972) |
| [docs/azure/sdk/authentication/local-development-dev-accounts.md](https://github.com/dotnet/docs/blob/d37b0d56517527ae281455e61fba4fcad98d19e2/docs/azure/sdk/authentication/local-development-dev-accounts.md) | [docs/azure/sdk/authentication/local-development-dev-accounts](https://review.learn.microsoft.com/en-us/dotnet/azure/sdk/authentication/local-development-dev-accounts?branch=pr-en-us-41972) |
| [docs/azure/sdk/authentication/local-development-service-principal.md](https://github.com/dotnet/docs/blob/d37b0d56517527ae281455e61fba4fcad98d19e2/docs/azure/sdk/authentication/local-development-service-principal.md) | [docs/azure/sdk/authentication/local-development-service-principal](https://review.learn.microsoft.com/en-us/dotnet/azure/sdk/authentication/local-development-service-principal?branch=pr-en-us-41972) |
| [docs/azure/sdk/authentication/on-premises-apps.md](https://github.com/dotnet/docs/blob/d37b0d56517527ae281455e61fba4fcad98d19e2/docs/azure/sdk/authentication/on-premises-apps.md) | [docs/azure/sdk/authentication/on-premises-apps](https://review.learn.microsoft.com/en-us/dotnet/azure/sdk/authentication/on-premises-apps?branch=pr-en-us-41972) |

### CODE REVIEW

**Review by CamSoper — APPROVED** [review]
LGTM, @scottaddie! You want me to merge or are you waiting on other reviews?

**Comment by CamSoper** [comment]
Oh wait, I'm not a codeowner on this, so my approval doesn't count. Weird. Looking into it.

**Comment by CamSoper** [comment]
@scottaddie FYI - https://github.com/dotnet/docs/pull/42016

File: docs/azure/sdk/authentication/additional-methods.md [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
 ms.topic: how-to
 ms.custom: devx-track-dotnet, engagement-fy23
-ms.date: 2/28/2023
+ms.date: 07/31/2024
 ---
 
 # Additional methods to authenticate to Azure resources from .NET apps
 
-This article lists additional methods apps may use to authenticate to Azure resources. The methods on this page are less commonly used and when possible, it is encouraged to use one of the methods outlined in the [authenticating .NET apps to Azure using the Azure SDK overview](./index.md) article.
+This article lists additional methods apps may use to authenticate to Azure resources. The methods on this page are less commonly used; when possible, we encourage you to use one of the methods outlined in [authenticating .NET apps to Azure using the Azure SDK overview](./index.md).
```
Comment (alexwolfmsft) [inline docs/azure/sdk/authentication/additional-methods.md:11]:
```suggestion
This article lists additional methods apps may use to authenticate to Azure resources. The methods on this page are less commonly used; when possible, use one of the methods outlined in [authenticating .NET apps to Azure using the Azure SDK overview](./index.md).
```

Optional nit: I don't think the "we" perspective/voice is usually used in Azure docs

File: docs/azure/sdk/authentication/azure-hosted-apps.md [hunk 2]
```diff
@@ -1,45 +1,45 @@
...
-description: This article covers how to configure authentication for apps to Azure services when the app is hosted in an Azure service like Azure App Service, Azure Functions, or Azure Virtual Machines.
+title: Authenticate Azure-hosted .NET apps to Azure resources
+description: Learn how to authenticate apps to Azure services when hosted in an Azure compute service like Azure App Service, Azure Functions, or Azure Virtual Machines.
 ms.topic: how-to
 ms.custom: devx-track-dotnet, engagement-fy23, devx-track-azurecli
-ms.date: 02/28/2023
+ms.date: 07/31/2024
 ---
 
 # Authenticating Azure-hosted apps to Azure resources with the Azure SDK for .NET
```
Comment (alexwolfmsft) [inline docs/azure/sdk/authentication/azure-hosted-apps.md:9]:
```suggestion
# Authenticate Azure-hosted apps to Azure resources with the Azure SDK for .NET
```

File: docs/azure/sdk/authentication/index.md [hunk 3]
```diff
@@ -1,56 +1,56 @@
...
-- **When an app is hosted on Azure -** The app should authenticate to Azure resources using a managed identity. This option is discussed in more detail below in the section [authentication in server environments](#authentication-in-server-environments).
-- **When an app is hosted and deployed on-premises -** The app should authenticate to Azure resources using an application service principal. This option is discussed in more detail below in the section [authentication in server environments](#authentication-in-server-environments).
+- **When a developer is running an app during local development**, the app can authenticate to Azure using either an application service principal for local development or by using the developer's Azure credentials. Each option is discussed in more detail at [authentication during local development](#authentication-during-local-development).
```
Comment (alexwolfmsft) [inline docs/azure/sdk/authentication/index.md:21]:
```suggestion
- **When an app is running locally during development**, the app can authenticate to Azure using either an application service principal for local development or by using the developer's Azure credentials. Each option is discussed in more detail at [authentication during local development](#authentication-during-local-development).
```

Consistency with the other list items.

File: docs/azure/sdk/authentication/index.md [hunk 4]
```diff
@@ -59,20 +59,21 @@ When an application is run on a developer's workstation during local development
...
-| Azure CLI                     | If a developer has authenticated to Azure using the `az login` command in the Azure CLI, `DefaultAzureCredential` will authenticate the app to Azure using that same account. |
-| Azure PowerShell              | If a developer has authenticated to Azure using the `Connect-AzAccount` cmdlet from Azure PowerShell, `DefaultAzureCredential` will authenticate the app to Azure using that same account.            |
+| Azure CLI                     | If a developer has authenticated to Azure using Azure CLI's `az login` command, `DefaultAzureCredential` will authenticate the app to Azure using that same account. |
+| Azure PowerShell              | If a developer has authenticated to Azure using Azure PowerShell's `Connect-AzAccount` cmdlet, `DefaultAzureCredential` will authenticate the app to Azure using that same account.            |
+| Azure Developer CLI           | If the developer has authenticated to Azure using Azure Developer CLI's `azd auth login` command, `DefaultAzureCredential` will authenticate with that account. |
```
Comment (alexwolfmsft) [inline docs/azure/sdk/authentication/index.md:78]:
```suggestion
| Azure Developer CLI           | If a developer has authenticated to Azure using Azure Developer CLI's `azd auth login` command, `DefaultAzureCredential` will authenticate with that account. |
```

Consistency with the other items

**Review by alexwolfmsft — APPROVED** [review]

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏

**Pull request merged**